File: packages/contracts/python/noa_contracts/floorplan_asset.py
from __future__ import annotations

from typing import Literal
from pydantic import BaseModel

from noa_contracts.common import ScaleInfo


class DimensionsPx(BaseModel):
    width: int
    height: int


class FloorplanAsset(BaseModel):
    schema_version: Literal["1.0"] = "1.0"
    id: str
    project_id: str
    source_type: Literal["pdf", "image", "cad_export", "scan", "hand_marked"]
    storage_key: str
    original_filename: str
    mime_type: str
    file_size_bytes: int
    dimensions_px: DimensionsPx | None
    page_count: int
    detected_scale: ScaleInfo | None
    upload_status: Literal["pending", "ready", "failed"]
    ingestion_job_id: str | None
    created_at: str

File: packages/contracts/v1/floorplan-asset.ts
import type { UUID, ISO8601 } from "./common.js";
import type { ScaleInfo } from "./common.js";

export interface FloorplanAsset {
  schema_version: "1.0";
  id: UUID;
  project_id: UUID;
  source_type: "pdf" | "image" | "cad_export" | "scan" | "hand_marked";
  storage_key: string;
  original_filename: string;
  mime_type: string;
  file_size_bytes: number;
  dimensions_px: { width: number; height: number } | null;
  page_count: number;
  detected_scale: ScaleInfo | null;
  upload_status: "pending" | "ready" | "failed";
  ingestion_job_id: UUID | null;
  created_at: ISO8601;
}
